### DISCUSSION TITLE
Quick Keys

### DISCUSSION THREAD

**Discussion by johnlindquist** [johnlindquist/kit#562] [Docs]
<meta path="help/quick-keys">

# Quick Keys

A quick key allows you to bind a single key to submit a prompt.

You can add quick keys inside the "hint" if you don't want to bother with choices:

```js
//Type "y" or "n"
let confirm = await arg({
  placeholder: "Eat a taco?",
  hint: `[y]es/[n]o`,
})

console.log(confirm) //"y" or "n"
```

Otherwise, add the quick keys in the `name` of the choices and it will return the quick key:

```js
 // Type "a", "b", or "g"
let fruit = await arg(`Pick one`, [
  `An [a]pple`,
  `A [b]anana`,
  `a [g]rape`,
])

console.log(fruit) //"a", "b", or "g"
```

You can add a value, then typing the quick key will return the value:

```js
// Type "c" or "a"
let vegetable = await arg("Pick a veggie", [
  { name: "[C]elery", value: "Celery" },
  { name: "C[a]rrot", value: "Carrot" },
])

console.log(vegetable) //"Celery" or "Carrot"
```

**Comment by nadeemkhedr** [comment]
@johnlindquist I love that, what do you think of expanding on it, so it wouldn't be just for args but also for script selection

I don't want to create a shortcut for every script, instead, I want to have one shortcut that opens up a menu and then press specific key(s) to run a script 

It would be very similar to `<leader>` and [which-key](https://github.com/folke/which-key.nvim) in vim

**Reply by johnlindquist** [reply]
You can use quick keys for scripts:

```js
// Menu: [R]un Me
```

The above will make your script trigger with "r". 

> 🚨 Note: The first script/choice _must_ have a quick key assigned or else it won't search for quick keys in the rest of the choices.

You can also build your own script menus. The "main" Script Kit menu is a script that uses this same logic/pattern:

```js
// Shortcut: option u
// You can only pick the script you want by filtering name/command/description/kenv/etc
let scriptFilter = scripts =>
  scripts.filter(script => script.name.includes("Update"))

let script = await selectScript(
  "Run script:", // The message to be displayed
  true, // Get scripts from cache
  scriptFilter // Transform scripts array
)
await run(script.command) // Run the script after selecting
```

**Reply by nadeemkhedr** [reply]
So cool! can you filter by getting only the quick key scripts? (Regex or there is a flag)

**Reply by johnlindquist** [reply]
As of now, it would have to be by regex.

I can look into adding it into the scripts db so we could filter by key though.

**Reply by nadeemkhedr** [reply]
Very excited about this, thanks for sharing that script, will play with it later
